src/System.AppContext/src/System/AppContext.cs

```diff
@@ -19,7 +19,6 @@ private enum SwitchValueState
         }
 
         private static Dictionary<string, SwitchValueState> s_switchMap = new Dictionary<string, SwitchValueState>();
-        private static readonly object s_syncLock = new object();
```

Comment:
Nit: Can you make s_switchMap readonly as well?  I rest easier when fields being used for synchronization can't be accidentally swapped out.